app/controllers/version.js

```diff
@@ -19,7 +19,13 @@ export default Controller.extend({
 
   actions: {
     selectVersion(version) {
-      this.transitionToRoute('version', version)
+      // Navigate to same section/page if it exists
+      const path = get(this, 'page.currentPage.url');
+      this.store.queryRecord('content', {version, path}).then(() => {
+        this.transitionToRoute(`/${version}/${path}/`);
+      }).catch(() => {
+        this.transitionToRoute('version', version);
+      })
```

Comment:
There is potential to reduce it to this:
```
  actions: {
    selectVersion(version) {
      this.transitionToRoute(`/${version}/${get(this, 'page.currentPage.url')}/`);
    }
  }
```
but changes would need to be made so that accessing an invalid path redirects back to the version index page, as right now this code just crashes.